components/PostAddForm.js

```diff
@@ -0,0 +1,70 @@
+import { useRouter } from 'next/router';
+import { useState, useCallback } from 'react';
+import Textarea from 'react-textarea-autosize';
+
+const PostAddForm = () => {
+  const { query: { offset } } = useRouter();
+  const [isExpanded, setExpanded] = useState(false);
+  const handleFocus = useCallback(() => setExpanded(true), []);
+  const handleCancel = useCallback(() => setExpanded(false), []);
+
+  if (offset)  {
+    return false;
+  }
```

Comment:
I don't like we create all those hooks up there only to `return false` here, feels like a waste. On the other hand, we can't just move this `if` to before the hooks, since that would break The Rule Of Hooks. So we can do the same trick here we did with PostLikes / PostLikesNotEmpty (see the beginning of `PostLikes.js`).

Reply:
It goes like this:

```
const PostLikes = () => {
  // 1. Only the hook we need to decide if we return early or not

  // 2. The IF with return

  // 3. return of NotEmpty component
};

const PostLikesNotEmpty = () => {
   // 4. The rest of the hooks

  // 5. Component "meat"
```